src/vector/_backends/numpy_.py

```diff
@@ -124,25 +122,25 @@ def _getitem(
 
         azimuthal, longitudinal, temporal = None, None, None
         if hasattr(array, "_azimuthal_type"):
-            azimuthal = array._azimuthal_type.ObjectClass(
+            azimuthal = array._azimuthal_type.ObjectClass(  # type: ignore
```

Comment:
Better typing on ObjectClass might fix this, it makes (wrong) assumptions about the constructor currently.